### PR TITLE
Patch 3

### PR DESCRIPTION
**Pull request by strider820**

### CODE REVIEW

File: bin/spotxBidAdapter.js [hunk 1]
```diff
@@ -20,28 +20,42 @@ export const spec = {
...
+    const videoMediaType = utils.deepAccess(bid, 'mediaTypes.video');
+    const context = utils.deepAccess(bid, 'mediaTypes.video.context');
+    if ((bid.mediaType === 'video' || (videoMediaType && context !== 'outstream')) || utils.getBidIdParameter('ad_unit', bid.params) == 'outstream') {
+      if (!utils.getBidIdParameter('outstream_function', bid.params)) {
+        if (!utils.getBidIdParameter('outstream_options', bid.params)) {
+          utils.logError(BIDDER_CODE + ': please define outstream_options parameter or override the default SpotX outstream rendering by defining your own Outstream function using field outstream_function.');
+          return false;
+        }
+        if (!utils.getBidIdParameter('video_slot', bid.params.outstream_options) || !utils.getBidIdParameter('content_width', bid.params.outstream_options) || !utils.getBidIdParameter('content_height', bid.params.outstream_options)) {
+          utils.logError(BIDDER_CODE + ': please define parameters video_slot, content_width and content_height for object outstream_options in the configuration.');
```
Comment (strider820) [inline bin/spotxBidAdapter.js:53]: Should read "please define parameters video_slot or content_width and content_height"

Also, as a side note, I think we need slot, not necessarily video_slot

File: bin/spotxBidAdapter.js [hunk 2]
```diff
@@ -324,15 +313,15 @@ function outstreamRender(bid) {
...
+      if (inIframe && window.document.getElementById(inIframe).nodeName == 'IFRAME') {
+        const rawframe = window.document.getElementById(inIframe);
         let framedoc = rawframe.contentDocument;
         if (!framedoc && rawframe.contentWindow) {
           framedoc = rawframe.contentWindow.document;
         }
         framedoc.body.appendChild(script);
       } else {
-        window.document.getElementById(bid.video_slot).appendChild(script);
+        window.document.getElementById(videoSlot).appendChild(script);
```
Comment (strider820) [inline bin/spotxBidAdapter.js:324]: yep, this should be a slot, not a video slot... videoslot is usually an actual video element

Comment (jmonbillard) [reply bin/spotxBidAdapter.js:324]: you right I am going to change this in the next push.

File: bin/spotxBidAdapter.js (hunk 1 above)
Comment (jmonbillard) [reply bin/spotxBidAdapter.js:53]: You are right. Let me push my proposal fixes for this.

**Pull request merged**